app/app/src/main/java/de/stustapay/stustapay/ui/payinout/topup/TopUpViewModel.kt

```diff
@@ -173,20 +175,11 @@ class TopUpViewModel @Inject constructor(
             }
         }
 
-        // TODO: maybe retry - but this should be done in http layer already...
-        when (val response = topUpRepository.bookTopUp(newTopUp)) {
-            is Response.OK -> {
-                clearDraft()
-                _topUpCompleted.update { response.data }
-                _status.update { "Card TopUp successful" }
-                _navState.update { TopUpPage.Done }
-            }
+        infallible.bookTopUp(newTopUp)
 
-            is Response.Error -> {
-                _status.update { "Card TopUp failed! ${response.msg()}" }
-                _navState.update { TopUpPage.Failure }
-            }
-        }
+        clearDraft()
+        _status.update { "Card TopUp successful" }
```

Comment:
what happens then the backend actually returns errors once the request gets through?
this should rather wait until the request was actually submitted and display the status afterwards.

Reply:
The backend can't return errors at this point. A transaction that has gotten this far (payment already completed via SumUp) may simply not fail.